persistence/src/main/java/com/epam/esm/dao/impl/GiftCertificateDaoImpl.java

```diff
@@ -1,144 +1,165 @@
-
-    private final SqlRequestParamParser parser;
-
-    @Autowired
-    public GiftCertificateDaoImpl(JdbcTemplate jdbcTemplate, SqlRequestParamParser parser) {
-        super(jdbcTemplate, GiftCertificate.class);
-        this.parser = parser;
-    }
+public class GiftCertificateDaoImpl extends AbstractDao<Long, GiftCertificate> implements GiftCertificateDao {
 
-    @Override
-    public List<GiftCertificate> findAll() {
-        return executeSelectQuery(FIND_ALL_SQL);
-    }
+    private static final String NAME_ATTRIBUTE = "name";
+    private static final String DESCRIPTION_ATTRIBUTE = "description";
+    private static final String TAGS_ATTRIBUTE = "tags";
 
-    @Override
-    public Optional<GiftCertificate> findById(Long id) {
-        return executeIdentifiableSelectQuery(FIND_BY_ID_SQL, id);
-    }
+    private final TagDao tagDao;
+    private final JpaUtil jpaUtil;
 
-    @Override
-    public void create(GiftCertificate entity) {
-        Long generatedId = executeInsertQueryReturnGeneratedKey(entity, Long.class);
-        entity.setId(generatedId);
+    public GiftCertificateDaoImpl(EntityManager entityManager, TagDao tagDao, JpaUtil jpaUtil) {
+        super(GiftCertificate.class, entityManager);
+        this.tagDao = tagDao;
+        this.jpaUtil = jpaUtil;
     }
 
     @Override
-    public boolean update(GiftCertificate entity) {
-        return executeUpdateQuery(entity);
+    protected void setNotNullFieldsToManagedEntity(GiftCertificate managedEntity, GiftCertificate newEntity) {
+        newEntity.getTags().stream()
+                .map(tagDao::createIfNotExists)
+                .forEach(managedEntity::addTag);
+        var name = Optional.ofNullable(newEntity.getName());
+        var description = Optional.ofNullable(newEntity.getDescription());
+        var price = Optional.ofNullable(newEntity.getPrice());
+        var duration = Optional.ofNullable(newEntity.getDuration());
+        var createDate = Optional.ofNullable(newEntity.getCreateDate());
+        var lastUpdateDate = Optional.ofNullable(newEntity.getLastUpdateDate());
+
+        name.ifPresent(managedEntity::setName);
+        description.ifPresent(managedEntity::setDescription);
+        price.ifPresent(managedEntity::setPrice);
+        duration.ifPresent(managedEntity::setDuration);
+        createDate.ifPresent(managedEntity::setCreateDate);
+        lastUpdateDate.ifPresent(managedEntity::setLastUpdateDate);
     }
 
     @Override
-    public boolean delete(Long id) {
-        return executeDeleteQuery(id);
+    public GiftCertificate create(GiftCertificate entity) {
+        List<Tag> tags = entity.getTags().stream()
+                .map(tagDao::createIfNotExists)
+                .toList();
+        entity.setTags(tags);
+        return super.create(entity);
     }
 
     @Override
-    protected String getTableName() {
-        return TABLE_NAME;
+    public Page<GiftCertificate> findAll(GiftCertificateFilters filters, Pageable pageable) {
```

Comment:
There is no way to understand how this method works. Maybe it would be better to write a native query?